### PR TITLE
methods are made for emai template and email confirigation using gmail smtp server 

### PR DESCRIPTION
**Pull request by praveenarjun**
## 📋 Description

# I used Mail nodemailer because normal sending and verifying mail is costly. In my code, the admin's Gmail has an option to send mail to anyone. I created the email templates and email generation.

## 🔨 Changes Made

 
- Created a folder in backend name mailtrap
 inside it I create email.js,emailTemplate,email trapconfig.js 
- 
- 

## ✅ Checklist

Before submitting the PR, please make sure you have completed the following:

export const MailtrapClient = nodemailer.createTransport({
    service: 'gmail', // or any other service like 'yahoo', 'outlook', etc.
    auth: {
        user: '', // your email address Generally to send mail it will be costly so u can add you gmail in gmail setting u can accept to send mail to any one and generate password
        pass: ' '    // your email password
    }
    need to add the name and password generated by smtp server of google 
     can refer this  ## https://www.youtube.com/watch?v=kTcmbZqNiGw link to generate 
     
     I only run my email verification because there is a issue in my docker desktop so Admin pls add Your mail and password in the place you can verify it
     
     Even I used Postman for api testing in the next pull request i will add the controller and router for this mail verification

## 🏷️ Types of Changes

What type of changes does your code introduce? (Check all that apply)

- [ ] New feature (non-breaking change which adds functionality) ✨
- [ ] UI enhancement (non-breaking change which enhances UI) 🎨

## 🤝 Related Issues

- Fixes #

## 📸 Screenshots
Screenshot 2024-10-05 at 10 40 53 PM

 (if app
Screenshot_20241005_222449_Gmail
licable)

---

## Reminder:

I created My own new Branch named as Email verification and email But I dont know how to pull request to your test branch here it is showing only main 
Please ensure your pull request is made to the `test` branch, not `main`.

*Thank you for contributing to VigneshDevHub! We look forward to reviewing your PR and getting it merged!*

### CODE REVIEW

File: backend/routes/userRoutes.js [hunk 1]
```diff
@@ -11,5 +11,7 @@ const router = express.Router();
 router.route("/").get(protect, allUsers);
 router.route("/").post(registerUser);
 router.post("/login", authUser);
+router.post("/verify-email", verifyEmail);
+router.post("/forgot-password", forgotPassword);
```
Comment (Sibam-Paul) [inline backend/routes/userRoutes.js:15]: Hey I think for forget password the issue is there #2 put for mail confirmation hmm.. wait let me see

Comment (Sibam-Paul) [reply backend/routes/userRoutes.js:15]: @praveenarjun Please add a short video of the changes u made and along with it's woking,  seeing fine with the code but with forget password having second thoughts

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]

**Comment by Sibam-Paul** [comment]
@praveenarjun U added methods in backend but I think for forget password #2 is working on it 
may cause contrast

**Comment by praveenarjun** [comment]
It may occur due to there need to generate a bearer token so that the token is generated and need to add a token time of expire by that only you can access the verification of email

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by Sibam-Paul** [comment]
@praveenarjun just dm me on discord my username specter102 or dm me in any general channel of gssoc-ext I need to see it live. as code is fine for me

**Comment by praveenarjun** [comment]
Ok I will but pls add a level in label pull request

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by Sibam-Paul** [comment]
@kartikayasija @kartikayasijaa yewala PR . Mujhe sahi lag raha hai .

**Comment by kartikayasijaa** [comment]
> @kartikayasija @kartikayasijaa yewala PR . Mujhe sahi lag raha hai .

Everything is working right?

**Comment by Sibam-Paul** [comment]
@praveenarjun Where I need to add my email ?? can u show me live in discord

**Comment by praveenarjun** [comment]
Yes but where is the discord link

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by praveenarjun** [comment]
IMG-20241007-WA0004
Here need to add the your mail and sender there is no need because it fetch from data base

**Pull request merged**